src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp

```diff
@@ -254,4 +254,42 @@ namespace OpenMS
     pp.pick(smoothed_spectrum, picked_spectrum);
     LOG_DEBUG << "Found " << picked_spectrum.size() << " spectrum peaks." << std::endl;
   }
+
+  void SpectrumExtractor::annotateSpectrum(
+    const std::vector<MSSpectrum>& spectra,
+    const TargetedExperiment& targeted_exp,
+    std::vector<MSSpectrum>& annotated_spectra
+  )
+  {
+    std::vector<ReactionMonitoringTransition> transitions = targeted_exp.getTransitions();
+    for (UInt i=0; i<spectra.size(); ++i)
+    {
+      MSSpectrum spectrum = spectra[i];
+      // It is supposed to have RT in minutes in target list file, therefore we divide by 60.0
+      const double spectrum_rt = spectrum.getRT() / 60.0;
+      const double rt_left_lim = spectrum_rt - getRTWindow() / 60.0 / 2.0;
+      const double rt_right_lim = spectrum_rt + getRTWindow() / 60.0 / 2.0;
+      const double spectrum_mz = spectrum.getPrecursors()[0].getMZ();
```

Comment:
Might be good to have a check to make sure that  `spectrum.getPrecursors()[0].getMZ();` actually has an mz value.  I am not sure if this will always be gauranteed.

Reply:
What should happen in case that information is absent for the some (or all) spectra? The method should keep filtering by the RT only or print an error and stop?